src/platform.rs

```diff
@@ -22,10 +22,10 @@ impl ToString for Platform {
 
 /// Read the current machine's platform.
 pub fn platform() -> Platform {
-    match env::consts::OS {
-        "linux" => Platform::LinuxAmd64,
-        "macos" => Platform::MacOsAmd64,
-        "windows" => Platform::WindowsAmd64,
+    match (env::consts::OS, env::consts::ARCH) {
+        ("linux", "x86_64") => Platform::LinuxAmd64,
+        ("macos", "x86_64") => Platform::MacOsAmd64,
```

Comment:
M1 macbook would have `env::consts::ARCH == aarch64`, so this would lead to an unsupported platform.

Although the mac OS binary works with Rosetta installed in the machine. I think we can ignore this check only for "macos", wdyt?

Reply:
I'm not sure. On one hand, I feel like we should be strict since we are getting reports in Foundry from M1 users without Rosetta caused by svm downloading x86_64 binaries. On the other hand, you should probably just have Rosetta while stuff is still adding support. I'll defer to you on this 😄

Reply:
I relaxed the requirement for now until we have a meaningful way to detect Rosetta in Rust or we get solc builds for M1